.jenkinsci/bindings.groovy

```diff
@@ -1,53 +1,104 @@
-  sh "zip -j $artifactsPath build/shared_model/bindings/*.java build/shared_model/bindings/libirohajava.so"
-  sh "cp $artifactsPath /tmp/bindings-artifact"
+  sh "cmake --build build --target irohajava --config ${buildType}"
+  // TODO: Java tests never finishes on Windows Server 2016. Disable until issue is resolved
+  // https://github.com/hyperledger/iroha/issues/1342
+  // sh "cd build; ctest -R java -C ${buildType} --output-on-failure"
+  sh "zip -j $artifactsPath build/bindings/*.java build/bindings/*.dll build/bindings/libirohajava.so"
+  if (os == 'windows') {
+    sh "cp $artifactsPath /tmp/${env.GIT_COMMIT}/bindings-artifact"
+  }
+  else {
+    sh "cp $artifactsPath /tmp/bindings-artifact"
+  }
   return artifactsPath
 }
 
-def doPythonBindings(buildType=Release) {
+def doPythonBindings(os, buildType=Release) {
   def currentPath = sh(script: "pwd", returnStdout: true).trim()
   def commit = env.GIT_COMMIT
   def supportPython2 = "OFF"
-  def artifactsPath = sprintf('%1$s/python-bindings-%2$s-%3$s-%4$s-%5$s.zip', 
-    [currentPath, env.PBVersion, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
-  // do not use preinstalled libed25519
-  sh "rm -rf /usr/local/include/ed25519*; unlink /usr/local/lib/libed25519.so; rm -f /usr/local/lib/libed25519.so.1.2.2"
+  def artifactsPath = sprintf('%1$s/python-bindings-%2$s-%3$s-%4$s-%5$s-%6$s.zip',
+    [currentPath, env.PBVersion, buildType, os, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
+  def cmakeOptions = ""
+  if (os == 'windows') {
+    sh "mkdir -p /tmp/${env.GIT_COMMIT}/bindings-artifact"
+    cmakeOptions = '-DCMAKE_TOOLCHAIN_FILE=/c/Users/Administrator/Downloads/vcpkg-master/vcpkg-master/scripts/buildsystems/vcpkg.cmake -G "NMake Makefiles"'
+  }
   if (env.PBVersion == "python2") { supportPython2 = "ON" }
   sh """
     cmake \
-      -H. \
+      -Hshared_model \
       -Bbuild \
       -DCMAKE_BUILD_TYPE=$buildType \
       -DSWIG_PYTHON=ON \
-      -DSUPPORT_PYTHON2=$supportPython2
+      -DSUPPORT_PYTHON2=$supportPython2 \
+      ${cmakeOptions}
   """
-  sh "cmake --build build --target python_tests"
-  sh "cd build; make -j${params.PARALLELISM} irohapy"
-  sh "protoc --proto_path=schema --python_out=build/shared_model/bindings block.proto primitive.proto commands.proto queries.proto responses.proto endpoint.proto"
-  sh "${env.PBVersion} -m grpc_tools.protoc --proto_path=schema --python_out=build/shared_model/bindings --grpc_python_out=build/shared_model/bindings endpoint.proto yac.proto ordering.proto loader.proto"
-  sh "zip -j $artifactsPath build/shared_model/bindings/*.py build/shared_model/bindings/*.so"
-  sh "cp $artifactsPath /tmp/bindings-artifact"
+  sh "cmake --build build --target irohapy --config ${buildType}"
+  sh "cmake --build build --target python_tests --config ${buildType}"
+  sh "cd build; ctest -R python -C ${buildType} --output-on-failure"
+  if (os == 'linux') {
```

Comment:
Why not ```if (os == 'windows') {...} else {...}```? It looks like this solution doesn't support MacOS anymore.

Reply:
Support for MacOS bindings with according code adjustments will be introduced later.